src/pebble.py

```diff
@@ -89,6 +89,14 @@ def stop_agent(self, container: ops.Container) -> None:
 
         Args:
             container: The agent workload container.
+
+        Raises:
+            APIError: if something went wrong with pebble requesting service stop.
         """
-        container.stop(self.state.jenkins_agent_service_name)
+        try:
+            container.stop(self.state.jenkins_agent_service_name)
+        except ops.pebble.APIError as exc:
+            if f'service "{self.state.jenkins_agent_service_name}" does not exist' in exc.message:
```

Comment:
Perhaps using `ops.pebble.APIError.code` to determine the type of error would be more resilient. I assume the error code here is `404`.

Reply:
The logs I've encountered didn't quite include the code, only the error message was displayed, I'll validate that the error code exists and apply the fix, thanks for the suggestion!

Reply:
@weiiwang01 I'm not quite confident on using `exc.code` in this case, the err code returned is 400 here and i think pebble might not be returning the error codes by convention - we might be catching errors outside of intended scope.
Log added: `logger.error("APIERR: %s %s", exc.message, exc.code)`
```
APIERR: cannot stop services: service "jenkins-k8s-agent" does not exist 400
```

Reply:
Gotcha 👍

Reply:
Isn't it more reliable to use `get_service` before calling stop, and only calling stop if the service is available ?
I find relying on the message of the exception to be a bit risky

Reply:
@gregory-schiano That is a great idea, fixed, thank you!